src/theme/DocItem/Footer/index.js

```diff
@@ -0,0 +1,206 @@
+  const buttonDislikeRef = useRef()
+
+  const [isTextboxVisible, setIsTextboxVisible] = useState(false)
+  const [isThanksVisible, setIsThanksVisible] = useState(false)
+
+  const handleLike = () => {
+    buttonLikeRef.current.blur()
+    setIsTextboxVisible(false)
+
+    setIsThanksVisible(true)
+    setTimeout(() => {
+      setIsThanksVisible(false)
+    }, 1000)
+
+    trackStructEvent({
+      category: 'feedback',
+      action: 'like',
+      label: permalink,
+    })
+  }
+
+  const handleDislike = () => {
+    buttonDislikeRef.current.blur()
+    setIsTextboxVisible((current) => !current)
+    setIsThanksVisible(false)
+
+    trackStructEvent({
+      category: 'feedback',
+      action: 'dislike',
+      label: permalink,
+    })
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    const text = feedbackTextRef.current.defaultValue
+
+    trackStructEvent({
+      category: 'feedback',
+      action: 'comment',
+      label: permalink,
+      property: text,
+    })
+
+    setIsTextboxVisible((current) => !current)
+    setIsThanksVisible(true)
+    setTimeout(() => {
+      setIsThanksVisible(false)
+    }, 1000)
+  }
+
+  return (
+    <div className="col margin-bottom--sm" style={{ paddingLeft: 0 + 'px' }}>
+      <div className="row margin-bottom--sm" style={{ marginLeft: 0 + 'px' }}>
+        <div className={styles.feedback_question}>Was this page helpful?</div>
+        <div>
+          <button
+            className={styles.feedback_button}
+            ref={buttonLikeRef}
+            onClick={handleLike}
+          >
+            Yes
+          </button>
+          <button
+            className={styles.feedback_button}
+            ref={buttonDislikeRef}
+            onClick={handleDislike}
+          >
+            No
+          </button>
+        </div>
+        {isTextboxVisible && (
+          <CommentBox
+            handleSubmit={handleSubmit}
+            feedbackTextRef={feedbackTextRef}
+          />
+        )}
+      </div>
+      <div className={clsx('row', styles.thanks)}>
+        {isThanksVisible && <span>Thanks for your feedback!</span>}
+      </div>
+    </div>
+  )
+}
+
+function EditMetaRow({
+  editUrl,
+  lastUpdatedAt,
+  lastUpdatedBy,
+  formattedLastUpdatedAt,
+}) {
+  return (
+    <div className={clsx(ThemeClassNames.docs.docFooterEditMetaRow, 'row')}>
+      <div className="col">
+        <Feedback />
+      </div>
+      <div style={{ marginLeft: 14 + 'px' }}>
+        <div className="col">
+          {editUrl && <EditThisPage editUrl={editUrl} />}
+        </div>
+
+        <div className={clsx('col', styles.lastUpdated)}>
+          {(lastUpdatedAt || lastUpdatedBy) && (
+            <LastUpdated
+              lastUpdatedAt={lastUpdatedAt}
+              formattedLastUpdatedAt={formattedLastUpdatedAt}
+              lastUpdatedBy={lastUpdatedBy}
+            />
+          )}
+        </div>
+      </div>
+    </div>
+  )
+}
+export default function DocItemFooter() {
```

Comment:
Would it be possible to wrap instead of ejecting? (`npm run swizzle @docusaurus/theme-classic DocItem/Footer -- --wrap --danger`) That way we can minimize the potential for breaking changes. The feedback widget could go right above or right below the footer.